python/lsst/resources/_resourceHandles/_httpResourceHandle.py

```diff
@@ -63,7 +65,11 @@ def fileno(self) -> int:
         raise io.UnsupportedOperation("HttpReadResourceHandle does not have a file number")
 
     def flush(self) -> None:
-        raise io.UnsupportedOperation("HttpReadResourceHandles are read only")
+        modes = set(self._mode.split())
+        if {"w", "x", "a", "+"} & modes:
+            raise io.UnsupportedOperation("HttpReadResourceHandles are read only")
+        if self._completeBuffer is not None:
+            self._completeBuffer.flush()
```

Comment:
This seems ok, but really it is just an implementation detail, flush on a read only handle is not supposed to do anything. I had set it to an unsupported operation because I didnt want people to think it did anything, but I think we mention in enough places handles are read only that it is fine.

Reply:
We have to make it be allowed. Forwarding on the flush seemed like the best way of being consistent but I can leave it as a no-op if you like.

Reply:
I think it is fine, it just that the _completeBuffer only is used in some code paths (when a full read is done) and I wanted to highlight that only in those paths is forwarding it on meaningful, perhaps a simple comment would be in order